core/pkg/ingress/types.go

```diff
@@ -0,0 +1,288 @@
+	// SSLCertificate path to the SSL certificate on disk
+	SSLCertificate string `json:"sslCertificate"`
+	// SSLPemChecksum returns the checksum of the certificate file on disk.
+	// There is no restriction in
+	// This helps to determine if the secret changed without the use of
+	// file system notifications
+	SSLPemChecksum string `json:"sslPemChecksum"`
+	// Locations list of URIs configured in the server
+	Locations []*Location `json:"locations,omitempty"`
+}
+
+// Location describes an URI inside a server.
+// Also contains additional information about defined annotations in
+// the Ingress rule.
+//
+// Important:
+// The implementation of annotations is optional
+type Location struct {
+	// Path is an extended POSIX regex as defined by IEEE Std 1003.1,
+	// (i.e this follows the egrep/unix syntax, not the perl syntax)
+	// matched against the path of an incoming request. Currently it can
+	// contain characters disallowed from the conventional "path"
+	// part of a URL as defined by RFC 3986. Paths must begin with
+	// a '/'. If unspecified, the path defaults to a catch all sending
+	// traffic to the backend.
+	Path string `json:"path"`
+	// IsDefBackend indicates if service specified in the Ingress
+	// contains active endpoints or not.
+	IsDefBackend bool `json:"isDefBackend"`
+	// Backend describes the endpoints to use
+	Backend Backend `json:"backend"`
+	// BasicDigestAuth returns authentication configuration for
+	// an Ingress rule
+	// +optional
+	BasicDigestAuth auth.BasicDigest `json:"basicDigestAuth,omitempty"`
+	// EnableCORS indicates if path must support CORS
+	// +optional
+	EnableCORS bool `json:"enableCors,omitempty"`
+	// ExternalAuth indicates the access to this location requires
+	// authentication using an external provider
+	// +optional
+	ExternalAuth authreq.External `json:"externalAuth,omitempty"`
+	// RateLimit describes
+	// +optional
+	RateLimit ratelimit.RateLimit `json:"rateLimit,omitempty"`
+	// Redirect describes the redirection this location
+	// +optional
+	Redirect rewrite.Redirect `json:"redirect,omitempty"`
+	// Whitelist indicates only connections from certain client
+	// addresses or networks are allowed
+	// +optional
+	Whitelist ipwhitelist.SourceRange `json:"whitelist,omitempty"`
+	// Proxy contains information about timeouts and buffer sizes
+	// to be used in connections against endpoints
+	// +optional
+	Proxy proxy.Configuration `json:"proxy,omitempty"`
+	// CertificateAuth indicates the access to this location requires
+	// external authentication
+	// +optional
+	CertificateAuth authtls.SSLCert `json:"certificateAuth,omitempty"`
+}
+
+// Backend describes one or more remote server/s (endpoints) associated with a service
+type Backend struct {
+	// Name represents an unique api.Service name formatted as <namespace>-<name>-<port>
+	Name string `json:"name"`
+	// This indicates if the communication protocol between the backend and the endpoint is HTTP or HTTPS
+	// Allowing the use of HTTPS
```

Comment:
If set to true, the ingress backend uses https for all communication to the endpoints (is this true?) including health checks. The endpoint needs to provide certificates with a valid hostname/ip (right? or is any self signed cert ok? if so please elaborate).